pkg/repository/v1/task.go

```
@@ -2578,16 +2609,27 @@ func (r *sharedRepository) createTaskEvents(
 	eventTypesStrs := make([]string, len(tasks))
 	paramDatas := make([][]byte, len(tasks))
 	paramKeys := make([]pgtype.Text, len(tasks))
+	externalIds := make([]pgtype.UUID, len(tasks))
 
 	internalTaskEvents := make([]InternalTaskEvent, len(tasks))
 
+	externalIdToData := make(map[pgtype.UUID][]byte, len(tasks))
+
 	for i, task := range tasks {
 		taskIds[i] = task.Id
 		taskInsertedAts[i] = task.InsertedAt
 		retryCounts[i] = task.RetryCount
 		eventTypesStrs[i] = string(eventTypes[i])
 
-		if len(eventDatas[i]) == 0 {
+		externalId := sqlchelpers.UUIDFromStr(uuid.NewString())
+		externalIds[i] = externalId
+
+		// important: if we don't set this to `eventDatas[i]` and instead allow it to be nil optionally
+		// we'll get errors downstream when we try to read the payload back and parse it in `registerChildWorkflows`
+		// because it'll try to unmarshal the `nil` value.
```

this is risky, but I'm not sure if we want to "fix" it downstream by swallowing that error or not 🤷